### ISSUE TITLE
Crash-[RCNConfigFetch refreshInstanceIDTokenAndFetchCheckInInfoWithCompletionHandler:]

### ISSUE THREAD

**Issue by kaushikpaperboat** [firebase/firebase-ios-sdk#5508]
### Step 0: Are you in the right place?

  * For issues or feature requests related to __the code in this repository__
    file a Github issue.
    * If this is a __feature request__ please use the Feature Request template.
  * For general technical questions, post a question on [StackOverflow](http://stackoverflow.com/)
    with the `firebase` tag.
  * For general (non-iOS) Firebase discussion, use the [firebase-talk](https://groups.google.com/forum/#!forum/firebase-talk)
    google group.
  * For backend issues, console issues, and other non-SDK help that does not fall under one
    of the above categories, reach out to
    [Firebase Support](https://firebase.google.com/support/).
  * Once you've read this section and determined that your issue is appropriate for
    this repository, please delete this section.

### [REQUIRED] Step 1: Describe your environment

  * Xcode version: 11.4.1
  * Firebase SDK version: 6.23.0
  * Firebase Component: Remote Config
  * Component version: 6.23.0
  * Installation method: CocoaPods

### [REQUIRED] Step 2: Describe the problem
We started seeing this crash after updating firebase to latest version.

#### Steps to reproduce:
We are not able to reproduce this issue in debug.
#### Crashlytics:
This is what we see on Crashlytics dashboard.
```
> Crashed: com.apple.main-thread
> 0  Kiddopia                       0x1dc702 __81-[RCNConfigFetch refreshInstanceIDTokenAndFetchCheckInInfoWithCompletionHandler:]_block_invoke_2 + 243 (RCNFetch.m:243)
> 1  Kiddopia                       0x1dc6ed __81-[RCNConfigFetch refreshInstanceIDTokenAndFetchCheckInInfoWithCompletionHandler:]_block_invoke_2 + 243 (RCNFetch.m:243)
> 2  Kiddopia                       0x1bfba7 __34-[FIRInstanceID getIDWithHandler:]_block_invoke + 450 (FIRInstanceID.m:450)
> 3  Kiddopia                       0x1b58a5 __49-[FIRInstallations installationIDWithCompletion:]_block_invoke + 179 (FIRInstallations.m:179)
> 4  Kiddopia                       0x206869 __56-[FBLPromise chainOnQueue:chainedFulfill:chainedReject:]_block_invoke.48 + 271 (FBLPromise.m:271)
> 5  libdispatch.dylib              0x23bf9823 _dispatch_call_block_and_release + 10
> 6  libdispatch.dylib              0x23bf980f _dispatch_client_callout + 22
> 7  libdispatch.dylib              0x23c07ba9 _dispatch_main_queue_callback_4CF$VARIANT$mp + 1524
> 8  CoreFoundation                 0x2404db6d __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 8
> 9  CoreFoundation                 0x2404c067 __CFRunLoopRun + 1574
> 10 CoreFoundation                 0x23f9b229 CFRunLoopRunSpecific + 520
> 11 CoreFoundation                 0x23f9b015 CFRunLoopRunInMode + 108
> 12 GraphicsServices               0x2558bac9 GSEventRunModal + 160
> 13 UIKit                          0x2866f189 UIApplicationMain + 144
> 14 Kiddopia                       0xbd8cd main + 14 (main.m:14)
> 15 ???                            0x23c43873 (Missing)
```

**Comment by maksymmalyhin** [comment]
@kaushikpaperboat Sorry you have the issue. It's a bit hard to understand what's wrong in your case from the information provided. I cannot see an obvious code error in the related sources. Could you provide the following information that may help with the further investigation: 

1. Call stack for other threads at the crash
2. Crash reason and exception content if available
3. Does the crash happen on a specific iOS version(s) or device(s)? 
4. Do you observe the crash with the previous Firebase versions?

NOTE: there are changes in the related code that are planned to be be included into the next release (currently in main branch). You may try to test this version from your side to check if the crash still occurs.

**Comment by kaushikpaperboat** [comment]
1. This is the complete call stack at the crash
```
#0 Crashed: com.apple.main-thread
0  Kiddopia                       0x19e586 __81-[RCNConfigFetch refreshInstanceIDTokenAndFetchCheckInInfoWithCompletionHandler:]_block_invoke_2 + 243 (RCNFetch.m:243)
1  Kiddopia                       0x19e571 __81-[RCNConfigFetch refreshInstanceIDTokenAndFetchCheckInInfoWithCompletionHandler:]_block_invoke_2 + 243 (RCNFetch.m:243)
2  Kiddopia                       0x181a43 __34-[FIRInstanceID getIDWithHandler:]_block_invoke + 450 (FIRInstanceID.m:450)
3  Kiddopia                       0x177741 __49-[FIRInstallations installationIDWithCompletion:]_block_invoke + 179 (FIRInstallations.m:179)
4  Kiddopia                       0x1c86c9 __56-[FBLPromise chainOnQueue:chainedFulfill:chainedReject:]_block_invoke.48 + 271 (FBLPromise.m:271)
5  libdispatch.dylib              0x23305823 _dispatch_call_block_and_release + 10
6  libdispatch.dylib              0x2330580f _dispatch_client_callout + 22
7  libdispatch.dylib              0x23313ba9 _dispatch_main_queue_callback_4CF$VARIANT$mp + 1524
8  CoreFoundation                 0x23759b6d __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 8
9  CoreFoundation                 0x23758067 __CFRunLoopRun + 1574
10 CoreFoundation                 0x236a7229 CFRunLoopRunSpecific + 520
11 CoreFoundation                 0x236a7015 CFRunLoopRunInMode + 108
12 GraphicsServices               0x24c97ac9 GSEventRunModal + 160
13 UIKit                          0x27d7b189 UIApplicationMain + 144
14 Kiddopia                       0x7e249 main + 14 (main.m:14)
15 (Missing)                      0x2334f873 (Missing)

#1. Thread
0  libsystem_kernel.dylib         0x23423864 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x234c9b35 _pthread_wqthread + 1036
2  libsystem_pthread.dylib        0x234c9718 start_wqthread + 8

#2. com.apple.libdispatch-manager
0  libsystem_kernel.dylib         0x234242f8 kevent_qos + 24
1  libdispatch.dylib              0x23319d61 _dispatch_mgr_invoke + 256
2  libdispatch.dylib              0x23319abf _dispatch_mgr_thread$VARIANT$mp + 38

#3. Thread
0  libsystem_kernel.dylib         0x23423864 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x234c9b35 _pthread_wqthread + 1036
2  libsystem_pthread.dylib        0x234c9718 start_wqthread + 8

#4. Thread
0  libsystem_kernel.dylib         0x23423864 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x234c9b35 _pthread_wqthread + 1036
2  libsystem_pthread.dylib        0x234c9718 start_wqthread + 8

#5. com.google.fira.worker
0  libsystem_kernel.dylib         0x2340f6e0 pread + 20
1  libsqlite3.dylib               0x23c228ad (null) + 2248
2  libsqlite3.dylib               0x23bbfe69 (null) + 11716
3  libsqlite3.dylib               0x23bd880f (null) + 4090
4  libsqlite3.dylib               0x23bd7bc1 (null) + 940
5  libsqlite3.dylib               0x23c30035 (null) + 57424
6  libsqlite3.dylib               0x23bf0c27 (null) + 38358
7  libsqlite3.dylib               0x23be7827 sqlite3_step + 470
8  Kiddopia                       0x4ca037 -[APMSqliteStore validateDatabaseWithError:] + 90
9  Kiddopia                       0x4c9f9d -[APMSqliteStore openAndValidateDatabase:] + 660
10 Kiddopia                       0x4c6005 -[APMSqliteStore initWithDatabasePath:error:] + 248
11 Kiddopia                       0x492287 -[APMDatabase initializeDatabaseResourcesWithContext:databasePath:error:] + 78
12 Kiddopia                       0x488d83 -[APMDatabase initWithDatabaseName:persistedConfig:error:] + 166
13 Kiddopia                       0x49a7cd __47-[APMMeasurement startMeasurementOnWorkerQueue]_block_invoke + 1196
14 Kiddopia                       0x49a201 -[APMMeasurement startMeasurementOnWorkerQueue] + 186
15 Kiddopia                       0x499e27 -[APMMeasurement setEnabledOnWorkerQueue:] + 292
16 Kiddopia                       0x499cf9 __29-[APMMeasurement setEnabled:]_block_invoke + 36
17 Kiddopia                       0x4b4a41 __51-[APMScheduler scheduleOnWorkerQueueBlockID:block:]_block_invoke + 92
18 libdispatch.dylib              0x23305823 _dispatch_call_block_and_release + 10
19 libdispatch.dylib              0x23312423 _dispatch_queue_drain$VARIANT$mp + 1758
20 libdispatch.dylib              0x23311a61 _dispatch_queue_invoke$VARIANT$mp + 284
21 libdispatch.dylib              0x2331415d _dispatch_root_queue_drain + 396
22 libdispatch.dylib              0x23313fcd _dispatch_worker_thread3 + 96
23 libsystem_pthread.dylib        0x234c9b29 _pthread_wqthread + 1024
24 libsystem_pthread.dylib        0x234c9718 start_wqthread + 8

#6. Thread
0  libsystem_kernel.dylib         0x23423864 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x234c9b35 _pthread_wqthread + 1036
2  libsystem_pthread.dylib        0x234c9718 start_wqthread + 8

#7. Thread
0  libsystem_kernel.dylib         0x23423864 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x234c9b35 _pthread_wqthread + 1036
2  libsystem_pthread.dylib        0x234c9718 start_wqthread + 8

#8. APMExperimentWorkerQueue
0  libsystem_kernel.dylib         0x23410430 fsync + 8
1  libsqlite3.dylib               0x23c0355f (null) + 36862
2  libsqlite3.dylib               0x23c09e05 (null) + 2084
3  libsqlite3.dylib               0x23c0335d (null) + 36348
4  libsqlite3.dylib               0x23bf7193 (null) + 13250
5  libsqlite3.dylib               0x23bd3fef (null) + 2422
6  libsqlite3.dylib               0x23bf3049 (null) + 47608
7  libsqlite3.dylib               0x23be7827 sqlite3_step + 470
8  libsqlite3.dylib               0x23bc137b sqlite3_exec + 402
9  Kiddopia                       0x4c975f -[APMSqliteStore endTransaction] + 44
10 Kiddopia                       0x4c9567 -[APMSqliteStore performTransactionWithError:block:] + 282
11 Kiddopia                       0x474577 -[APMEDatabase performTransaction:] + 44
12 Kiddopia                       0x47c5d7 -[APMETaskManager fetchAllSnapshotsOnWorkerQueue] + 134
13 Kiddopia                       0x47c535 -[APMETaskManager startTaskManagerOnWorkerQueue] + 144
14 Kiddopia                       0x47c49b __35-[APMETaskManager startTaskManager]_block_invoke + 30
15 Kiddopia                       0x47da0b __46-[APMETaskManager dispatchAsyncOnWorkerQueue:]_block_invoke + 90
16 libdispatch.dylib              0x23305823 _dispatch_call_block_and_release + 10
17 libdispatch.dylib              0x23312423 _dispatch_queue_drain$VARIANT$mp + 1758
18 libdispatch.dylib              0x23311a61 _dispatch_queue_invoke$VARIANT$mp + 284
19 libdispatch.dylib              0x2331415d _dispatch_root_queue_drain + 396
20 libdispatch.dylib              0x23313fcd _dispatch_worker_thread3 + 96
21 libsystem_pthread.dylib        0x234c9b29 _pthread_wqthread + 1024
22 libsystem_pthread.dylib        0x234c9718 start_wqthread + 8

#9. com.twitter.crashlytics.ios.MachExceptionServer
0  Kiddopia                       0x4496a1 CLSProcessRecordAllThreads + 152
1  Kiddopia                       0x4496a1 CLSProcessRecordAllThreads + 152
2  Kiddopia                       0x449931 CLSProcessRecordAllThreads + 808
3  Kiddopia                       0x43d005 CLSHandler + 30
4  Kiddopia                       0x43949f CLSMachExceptionServer + 1158
5  libsystem_pthread.dylib        0x234cb85b _pthread_body + 138
6  libsystem_pthread.dylib        0x234cb7cf _pthread_start + 110
7  libsystem_pthread.dylib        0x234c9724 thread_start + 8

#10. Thread
0  libsystem_kernel.dylib         0x23423864 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x234c9b35 _pthread_wqthread + 1036
2  libsystem_pthread.dylib        0x234c9718 start_wqthread + 8

#11. Thread
0  libsystem_kernel.dylib         0x23423864 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x234c9b35 _pthread_wqthread + 1036
2  libsystem_pthread.dylib        0x234c9718 start_wqthread + 8

#12. Thread
0  libsystem_kernel.dylib         0x23423864 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x234c9b35 _pthread_wqthread + 1036
2  libsystem_pthread.dylib        0x234c9718 start_wqthread + 8

#13. Thread
0  libsystem_kernel.dylib         0x23423864 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x234c9b35 _pthread_wqthread + 1036
2  libsystem_pthread.dylib        0x234c9718 start_wqthread + 8

#14. com.apple.NSURLConnectionLoader
0  libsystem_kernel.dylib         0x2340e8d0 mach_msg_trap + 20
1  libsystem_kernel.dylib         0x2340e6d5 mach_msg + 40
2  CoreFoundation                 0x23759ac5 __CFRunLoopServiceMachPort + 136
3  CoreFoundation                 0x23757e4d __CFRunLoopRun + 1036
4  CoreFoundation                 0x236a7229 CFRunLoopRunSpecific + 520
5  CoreFoundation                 0x236a7015 CFRunLoopRunInMode + 108
6  CFNetwork                      0x23cf89f7 +[NSURLConnection(Loader) _resourceLoadLoop:] + 486
7  Foundation                     0x23fc14a5 __NSThread__start__ + 1148
8  libsystem_pthread.dylib        0x234cb85b _pthread_body + 138
9  libsystem_pthread.dylib        0x234cb7cf _pthread_start + 110
10 libsystem_pthread.dylib        0x234c9724 thread_start + 8

#15. com.apple.CFSocket.private
0  libsystem_kernel.dylib         0x23422eec __select + 20
1  CoreFoundation                 0x2375ef13 __CFSocketManager + 566
2  libsystem_pthread.dylib        0x234cb85b _pthread_body + 138
3  libsystem_pthread.dylib        0x234cb7cf _pthread_start + 110
4  libsystem_pthread.dylib        0x234c9724 thread_start + 8

#16. AVAudioSession Notify Thread
0  libsystem_kernel.dylib         0x2340e8d0 mach_msg_trap + 20
1  libsystem_kernel.dylib         0x2340e6d5 mach_msg + 40
2  CoreFoundation                 0x23759ac5 __CFRunLoopServiceMachPort + 136
3  CoreFoundation                 0x23757e4d __CFRunLoopRun + 1036
4  CoreFoundation                 0x236a7229 CFRunLoopRunSpecific + 520
5  CoreFoundation                 0x236a7015 CFRunLoopRunInMode + 108
6  libAVFAudio.dylib              0x29066225 GenericRunLoopThread::Entry(void*) + 132
7  libAVFAudio.dylib              0x2903b177 CAPThread::Entry(CAPThread*) + 154
8  libsystem_pthread.dylib        0x234cb85b _pthread_body + 138
9  libsystem_pthread.dylib        0x234cb7cf _pthread_start + 110
10 libsystem_pthread.dylib        0x234c9724 thread_start + 8

#17. AURemoteIO::IOThread
0  libsystem_kernel.dylib         0x2340e8d0 mach_msg_trap + 20
1  libsystem_kernel.dylib         0x2340e6d5 mach_msg + 40
2  AudioToolbox                   0x25be41d9 AURemoteIO::IOThread::Run() + 104
3  AudioToolbox                   0x25be7a89 AURemoteIO::IOThread::Entry(void*) + 8
4  AudioToolbox                   0x25b005af CAPThread::Entry(CAPThread*) + 186
5  libsystem_pthread.dylib        0x234cb85b _pthread_body + 138
6  libsystem_pthread.dylib        0x234cb7cf _pthread_start + 110
7  libsystem_pthread.dylib        0x234c9724 thread_start + 8

#18. Thread
0  libsystem_kernel.dylib         0x23423864 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x234c9b35 _pthread_wqthread + 1036
2  libsystem_pthread.dylib        0x234c9718 start_wqthread + 8

#19. Thread
0  libsystem_kernel.dylib         0x23423864 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x234c9b35 _pthread_wqthread + 1036
2  libsystem_pthread.dylib        0x234c9718 start_wqthread + 8

```
2. Crash reason and exception content if available
On Crashlytics it says EXC_BAD_ACCESS KERN_INVALID_ADDRESS.

3. Does the crash happen on a specific iOS version(s) or device(s)?
This crash we are seeing on almost all iOS Devices and all iOS versions.As we are doing phased release we don't have proper data to determine anything.

4. Do you observe the crash with the previous Firebase versions?
It was working fine before updating firebase.

**Comment by maksymmalyhin** [comment]
@kaushikpaperboat Thank you for the update. The particular reason of the crash is still not clear, but recently released Firebase 6.24.0 contains refactoring of related logic, so potentially the crash might be fixed there. Could you please try to update Firebase to 6.24.0 and check if you still have a crash?
